Windows/Debugger/CtrlDisAsmView.cpp

```diff
@@ -1377,3 +1542,50 @@ u32 CtrlDisAsmView::getInstructionSizeAt(u32 address)
 	u32 next  = manager.getNthNextAddress(start,1);
 	return next-address;
 }
+
+int regStringToMIPSGPReg(const char* name) {
```

Comment:
Here you may want to create a new struct that have both the int and the string.

```
struct regStringPair {
    int reg;
    const char *name;
};
```

And then declare and array with all the pair:

```
const regStringPair regNames[] = {
    {MIPS_REG_V0, "vo"},
    {MIPS_REG_V1, "v1"},
    ....
};
```

Then just loop the array and if you have match return what you want.

Reply:
I think i will try this. And maybe move it outside mips.h, too.

Reply:
Yeah, way better.

Just a few thing:
You may just change the parameters type from `const char *name` to `const std::string &name` and then in the loop you can check with `if (name == regNames[i].name)`.

After this no need to add `.c_str()` when you call this function.

When you check the return type of this check for MIPS_REG_INVAID rather than -1, it get more clear what you are doing like that.

I would change the loop with `i < ARRAY_SIZE(regNames)`, so you don't have to count them and even if you change the array it will still work.

Reply:
Temporarily unresolving because i didn't realize the ARRAY_SIZE function existed and i will try to incorporate that as soon as i next have spoons to look at this.

Reply:
Just to note ARRAY_SIZE is not a function but rather a macro, it get expanded by preprocessor to something like:
`sizeof(array)/sizeof(array[0])` and doesn't work on dynamic array or pointer at all. Nothing to concern about here, just for future reference.